storage/src/vespa/storage/distributor/distributor_bucket_space.cpp

```diff
@@ -114,19 +114,40 @@ DistributorBucketSpace::owns_bucket_in_state(
     return owns_bucket_in_state(*_distribution, clusterState, bucket);
 }
 
+namespace {
+
+void
+setup_ideal_nodes_bundle(IdealServiceLayerNodesBundle& ideal_nodes_bundle,
+                         const lib::Distribution& distribution,
+                         const lib::ClusterState& cluster_state,
+                         document::BucketId bucket)
+{
+    ideal_nodes_bundle.set_available_nodes(distribution.getIdealStorageNodes(cluster_state, bucket, up_states));
+    ideal_nodes_bundle.set_available_nonretired_nodes(distribution.getIdealStorageNodes(cluster_state, bucket, nonretired_up_states));
+    ideal_nodes_bundle.set_available_nonretired_or_maintenance_nodes(distribution.getIdealStorageNodes(cluster_state, bucket, nonretired_or_maintenance_up_states));
+}
+
+// Ideal service layer nodes bundle used when bucket id used bits > 33.
+thread_local IdealServiceLayerNodesBundle fallback_ideal_nodes_bundle;
+
+}
+
 const IdealServiceLayerNodesBundle&
 DistributorBucketSpace::get_ideal_service_layer_nodes_bundle(document::BucketId bucket) const
 {
     assert(bucket.getUsedBits() >= _distribution_bits);
+    if (bucket.getUsedBits() > 33) {
```

Comment:
Consider adding a constexpr variable for this to make it clearer. It is also used elsewhere (e.g. L143). Would be nice with a comment on why 33 is selected as well.